lib/action_cable/connection/base.rb

```diff
@@ -165,6 +165,28 @@ def on_close
         end
 
 
+        def allow_request_origin?
+          return true if server.config.disable_request_forgery_protection
+
+          if env['HTTP_ORIGIN'].present?
+            origin_host = URI.parse(env['HTTP_ORIGIN']).host
+
+            allowed = if server.config.allowed_request_origins.present?
+              Array.wrap(server.config.allowed_request_origins).include? origin_host
+            else
+              request.host == origin_host
```

Comment:
This is a sensible default, but needs closer consideration. We don't whitelist the Host either, so a matching Host and Origin could allow someone to bypass our Origin checks. Clearer to always require a whitelist entry.